src/evo/deterministicmns.cpp

```diff
@@ -131,8 +131,7 @@ bool CDeterministicMNList::IsMNValid(const CDeterministicMNCPtr &dmn, int height
 }
 
 bool CDeterministicMNList::IsMNValid(const CDeterministicMNCPtr &dmn) {
-    int height = ::ChainActive().Tip() == nullptr ? 0 : ::ChainActive().Tip()->nHeight;
-    return IsMNValid(dmn, height);
+    return IsMNValid(dmn, ::ChainActive().AtomicHeight());
```

Comment:
The non-core code (i.e., Qt) is calling IsMNValid without a lock to cs_main.  Because of this, the call to .Tip() is not safe.  Switched to the AtomicHeight which was created for this very reason.